client/src/components/web-extension/Common/Header/OuterHeader.jsx

```diff
@@ -1,38 +1,34 @@
 import React, { useCallback, useMemo } from 'react';
 import { useNavigate, useLocation } from 'react-router-dom';
-import { useSelector, useDispatch } from 'react-redux';
-import { selectCreateWalletCurrentStep } from '@cd/selectors/createWallet';
+import { useDispatch } from 'react-redux';
 import { resetWalletCreation } from '@cd/actions/createWalletActions';
 import BackArrow from '@cd/assets/image/back-arrow.svg';
 import './OuterHeader.scss';
 
 export const OuterHeader = ({ setHeader, header }) => {
 	const dispatch = useDispatch();
-	const currentStep = useSelector(selectCreateWalletCurrentStep);
 	const navigate = useNavigate();
 	const { pathname, state } = useLocation();
 	const shouldShowBackArrow = pathname !== '/welcomeBack';
 
 	const finalLayoutName = useMemo(() => {
-		if (header && header !== '') {
-			return header;
-		}
-
 		if (!state?.name) {
 			return undefined;
 		}
 
+		if (header && header !== '') {
+			return header;
+		}
+
```

Comment:
I moved the conditional check. `state` data becomes `undefined` after navigating to /connectAccount screen from inner child routes.